crates/bevy_render/src/camera/projection.rs

```diff
@@ -112,3 +112,84 @@ impl Default for OrthographicProjection {
         }
     }
 }
+
+#[derive(Debug, Clone, Reflect, Serialize, Deserialize)]
+#[reflect_value(Serialize, Deserialize)]
+pub enum BaseAxis {
+    Vertical,
+    Horizontal,
+}
+
+#[derive(Debug, Clone, Reflect)]
+#[reflect(Component)]
+pub struct ScaledOrthographicProjection {
+    pub scale: f32,
+    pub aspect_ratio: f32,
+    pub near: f32,
+    pub far: f32,
+    pub window_origin: WindowOrigin,
+    pub base_axis: BaseAxis,
+}
+
+impl CameraProjection for ScaledOrthographicProjection {
+    fn get_projection_matrix(&self) -> Mat4 {
+        match (&self.window_origin, &self.base_axis) {
+            (WindowOrigin::Center, BaseAxis::Vertical) => Mat4::orthographic_rh(
+                -self.aspect_ratio * self.scale,
+                self.aspect_ratio * self.scale,
+                -self.scale,
+                self.scale,
+                self.near,
+                self.far,
+            ),
+            (WindowOrigin::BottomLeft, BaseAxis::Vertical) => Mat4::orthographic_rh(
+                0.0,
+                self.aspect_ratio * self.scale,
+                0.0,
+                self.scale,
+                self.near,
+                self.far,
+            ),
+            (WindowOrigin::Center, BaseAxis::Horizontal) => Mat4::orthographic_rh(
+                -self.scale,
+                self.scale,
+                -self.aspect_ratio * self.scale,
+                self.aspect_ratio * self.scale,
+                self.near,
+                self.far,
+            ),
+            (WindowOrigin::BottomLeft, BaseAxis::Horizontal) => Mat4::orthographic_rh(
+                0.0,
+                self.scale,
+                0.0,
+                self.aspect_ratio * self.scale,
+                self.near,
+                self.far,
+            ),
+        }
+    }
+
+    fn update(&mut self, width: f32, height: f32) {
+        self.aspect_ratio = match self.base_axis {
+            BaseAxis::Vertical => width / height,
+            BaseAxis::Horizontal => height / width,
+        }
+    }
+
+    fn depth_calculation(&self) -> DepthCalculation {
+        DepthCalculation::ZDifference
+    }
+}
+
+impl Default for ScaledOrthographicProjection {
+    fn default() -> Self {
+        ScaledOrthographicProjection {
+            scale: 1.0,
```

Comment:
I think this scale field shouldn't be here, and instead should be handled by the scale on the `Transform` of `self`.

Obviously that would also scale `far`, which isn't ideal.

Reply:
I strongly disagree. They serve different purposes. I think it is important to have this field, to be able to set the "span" of the projection.

The camera's transform is typically used to perform "zoom in/out" effects in the game. The "scale" of the transform is if you need to normalize the sizes of things (say, depending on what units you work in, particularly for a 2D game).

Say, for example, you are making a 2D game and you want it to scale based on screen resolution, but always fit the same content on screen.

You might want the projection to go -600.0 to +600.0 (1200 total height), because you designed your assets for a nominal 1200-vertical-pixels screen size. You shouldn't have to complicate your camera math to account for that. The camera scale should stay at 1.0 unless you are doing some "zoom in/out" effects.

This is the same argument I've been giving before w.r.t your suggestion of "just use the window-pixels 2d projection bevy already provides for 2D games and fiddle with the camera scale to make your content fit the screen" ... it's an ugly workaround and not ergonomic.

It really doesn't cost anything to have this field, and I think it opens up some likely-useful extra flexibility for users. Why potentially cripple/limit the functionality, just to avoid that extra field? IMO it makes sense to have it.

Reply:
My point is that if we did have scale only as the field on transform, then that would be the only thing defining the size (of the fixed axis) of the in-world screen/size of the in-world viewing pane

Whereas at the moment you have to multiply it by this scale field to know that

Reply:
Well, the projection scale is at 1.0 by default, so what you said applies, unless the user explicitly goes to mess with the projection. I think it is valuable for the users to have the option of doing that.

Which means we shouldn't remove the scale field.

It's not going to surprise or confuse anyone; it's only relevant to those users who know they want to use it.

Reply:
Yeah, that also makes sense.